packages/opentelemetry-tracing/src/BasicTracerProvider.ts

```diff
@@ -80,6 +96,18 @@ export class BasicTracerProvider implements TracerProvider {
    * @param spanProcessor the new SpanProcessor to be added.
    */
   addSpanProcessor(spanProcessor: SpanProcessor): void {
+    if (this._registeredSpanProcessors.length === 0) {
+      // since we might have enabled by default a batchProcessor, we disable it
+      // before adding the new one
+      this.activeSpanProcessor
+        .shutdown()
```

Comment:
I don't think it's important since the following code is sync, the property `this.activeSpanProcessor` is updated just and there are no way a span could be exporter by the old AND the new span processor.
This also avoid problem if the previous span processor is taking a lot of time to shutdown